mobile-browser-based-version/mobile-app/src/components/TitanicComponents/titanic_training.vue

```diff
@@ -0,0 +1,1018 @@
+                    wrongRow
+                  )
+                );
+              } else {
+                alert(
+                  "Training aborted. The following columns name were not found in the uploaded file: ".concat(
+                    wrongRow
+                  )
+                );
+              }
+            }
+          }
+
+          var startTraining = checkHeaderLength && checkHeaderContent;
+          // If user's file respects our format, parse it and start training
+          if (startTraining) {
+            console.log("User File Validated. Start parsing.");
+            console.log(headerCopied);
+            let Xcsv = d3.csvParse(
+              e.target.result,
+              function (d) {
+                return [
+                  +d[headerCopied[0]], // PassengerId
+                  +d[headerCopied[1]], // Survived
+                  d[headerCopied[3]] == "male" ? 1 : 0, // Sex
+                  +d[headerCopied[4]], // Age
+                  +d[headerCopied[5]], // SibSp
+                  +d[headerCopied[6]], // Parch
+                  +d[headerCopied[8]], // Fare
+                  +d[headerCopied[11]], // Pclass
+                ];
+              },
+              function (error, rows) {
+                console.log(rows);
+              }
+            );
+
+            let ycsv = d3.csvParse(
+              e.target.result,
+              function (d) {
+                return +d[headerCopied[1]];
+              },
+              function (error, rows) {
+                console.log(rows);
+              }
+            );
+            /*
+                return {
+                    // Map the user's data point to our format 
+                    // and convert what can be converted to int or string (with +) 
+                    PassengerId: +d[headerCopied[0]],
+                    Survived: +d[headerCopied[1]],
+                    Name: d[headerCopied[2]],
+                    Sex: d[headerCopied[3]],
+                    Age: +d[headerCopied[4]],
+                    SibSp: +d[headerCopied[5]],
+                    Parch: +d[headerCopied[6]],
+                    Ticket: d[headerCopied[7]],
+                    Fare: +d[headerCopied[8]],
+                    Cabin: d[headerCopied[9]],
+                    Embarked: d[headerCopied[10]],
+                    Pclass: +d[headerCopied[11]]
+                };*/
+
+            // console.log(tf.tensor2d(Xcsv));
+            // console.log(tf.tensor1d(ycsv));
+
+            let Xtrain = tf.tensor2d(Xcsv);
+            let ytrain = tf.tensor1d(ycsv);
+
+            //let df = new DataFrame(parsedCSV)
+            // console.log(df)
+            this.train(Xtrain, ytrain);
+          } else {
+            console.log("Cannot Start training");
+          }
+        };
+
+        reader.readAsText(file);
+      }
+    },
+
+    /**
+     * Creates the tf.model
+     * TO DO: fetch model from local storage
```

Comment:
clarify that this is architecture + model weights (initialized to zero for now. for future better warm-start, we can let the train loop take care of it).  load + save to local storage can be implemented later, no prob